doc/man/man3/crypto_blake2b.3monocypher

```diff
@@ -276,3 +281,18 @@ Any deviation from the specified input and output length ranges results
 in
 .Sy undefined behaviour .
 Make sure your inputs are correct.
+.Sh SECURITY CONSIDERATIONS
+BLAKE2b is a general-purpose cryptographic hash function;
+this means that it is not suited for hashing passwords and deriving
```

Comment:
For the average coder it would be helpful to state first some example what Blake3b is *suitable* for before telling what it is *not suitable* for. Just for a clear separation.

Reply:
I'm not really sure how to approach this issue, both from a "How do I avoid writing a book on applied cryptography in the form of a sequence of man pages?" perspective, and from a "What is actually useful for most people to know?" The problem with explaining that is that there is too much it *is* useful for. Applications include:

- checking if data is identical (e.g. quick file sameness comparison)
- checking if data is unchanged from some known-good value (though usually you'll want signtures or MACs instead), see also hash trees for doing this at scale
- *non-password* key derivation (as required in some protocols, IIRC PURBs make use of this)
- as a core component of signatures (but Monocypher figures that out for you as part of `crypto_sign()`, so you'll never actually have to touch this yourself)
- as part of random number generators (but it seems everyone's moved to stream ciphers instead)
- message authentication codes, i.e. symmetric signatures (BLAKE2b in keyed mode; SHA-512 with HMAC); though the manual actually mentions this one when talking about the `key` argument
- as part of PBKDF2 (but please don't; Monocypher has Argon2)
- as part of key agreement (hashing in the conversation until a certain point, cf. Monokex)
- ...

Most of these are only necessary for implementing more elaborate constructions yourself and I feel like this list just creates more questions than it answers. I'm not sure what you're looking for here. See also the [NaCl documentation, which categorizes hash functions as "low-level functions"](https://nacl.cr.yp.to/hash.html) instead.

CC @LoupVaillant since intro(3monocypher) also tells people to read Crypto101.